server/controllers/signupControllers/signup.js

```diff
@@ -0,0 +1,27 @@
+const { createCookies, createSession } = require('../../utils');
+const checkUser = require('./checkUser');
+const createUser = require('./createUser');
+const validateSignup = require('./validation');
+
+const signup = (req, res) => {
+  const {
+    userName, email, firstName, password,
+  } = req.body;
+  const validation = validateSignup(userName, email, firstName, password);
+  if (validation) { return res.status(403).json({ message: validation }); }
+
+  checkUser(userName, email).then((exists) => {
+    // eslint-disable-next-line no-throw-literal
+    if (exists) { throw { message: exists, cause: 'validation error' }; }
+  }).then(() => createUser(userName, email, firstName, password))
+    .then(() => {
+      createCookies(res, userName, true, createSession(userName));
+      res.redirect('/');
```

Comment:
While we are working with react, and our routes handeled in frontend, Just send a JSON response with a detailed message and take care that after adding new user Status shouls be 201 not 200,
For more information https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201